### ISSUE TITLE
Signals and Objects lists are very unwieldy to use

### ISSUE THREAD

**Issue by jerkstorecaller** [KDAB/GammaRay#647]
Signals and Objects are gigantic lists which are 99.9% Qt classes. Finding my own code is like looking for a needle in a haystack. It would be nice to be able to configure an exclusion list. And I would add a checkbox toggle "exclude Qt classes" as an existing filter.

Alternatively (or perhaps on top of the above), the search/filter inputbox at the top should accept multiple terms. Like how grep lets you match on multiple words using "word1\|word2\|word3" (Github edited out the forward slash, you know what I mean), I wish I could search for "Class1 OR Class2 OR Class3".

Basically, ANYTHING to be able to look at my own objects, without having to search for each one individually and being unable to alternate between 2 objects quickly.

**Comment by Waqar144** [comment]
https://github.com/KDAB/GammaRay/pull/704
